src/pudl/transform/eia930.py

```diff
@@ -0,0 +1,224 @@
+        .astype(
+            {
+                "balancing_authority_code_eia": pd.CategoricalDtype(),
+                "energy_source": pd.CategoricalDtype(),
+            }
+        )
+    )
+    # TODO[zaneselvans] 2024-04-20: Verify that sum of net generation from all fuels
+    # adds up to the total And then drop the total rows.
+    # NOTE: currently there are some BIG differences between the calculated total and
+    # the reported total.
+
+    # netgen_stacked = netgen_stacked[netgen_stacked["energy_source"] != "total"
+
+    demand = raw_eia930__balance[
+        qual_cols + list(raw_eia930__balance.filter(like="demand"))
+    ].astype({"balancing_authority_code_eia": pd.CategoricalDtype()})
+
+    return (
+        Output(
+            value=netgen_stacked,
+            output_name="core_eia930__hourly_balancing_authority_net_generation",
+        ),
+        Output(
+            value=demand,
+            output_name="core_eia930__hourly_balancing_authority_demand",
+        ),
+    )
+
+
+@multi_asset(
+    outs={
+        "core_eia930__hourly_subregion_demand": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+        "core_eia930__assn_balancing_authority_subregion": AssetOut(
+            io_manager_key="pudl_io_manager"
+        ),
+    },
+    compute_kind="pandas",
+)
+def core_eia930__hourly_subregion_assets(raw_eia930__subregion: pd.DataFrame):
+    """Produce a normalized table of hourly demand by subregion."""
+    demand = (
+        raw_eia930__subregion.assign(
+            subregion_code_eia=lambda df: df["subregion_code_eia"].str.upper()
+        )
+        .astype(
+            {
+                "balancing_authority_code_eia": pd.CategoricalDtype(),
+                "subregion_code_eia": pd.CategoricalDtype(),
+            }
+        )
+        .loc[
+            :,
+            [
+                "report_datetime_local",
+                "report_datetime_utc",
+                "balancing_authority_code_eia",
+                "subregion_code_eia",
+                "demand_reported_mw",
+            ],
+        ]
+    )
+    assn = (
+        demand.groupby("balancing_authority_code_eia")["subregion_code_eia"]
+        .unique()
+        .explode()
+        .to_frame()
+        .reset_index()
+        .astype(
+            {
+                "balancing_authority_code_eia": pd.CategoricalDtype(),
+                "subregion_code_eia": pd.CategoricalDtype(),
+            }
+        )
+    )
+    return (
+        Output(value=demand, output_name="core_eia930__hourly_subregion_demand"),
+        Output(
+            value=assn, output_name="core_eia930__assn_balancing_authority_subregion"
+        ),
+    )
+
+
+@multi_asset(
+    outs={
+        "core_eia930__hourly_balancing_authority_interchange": AssetOut(
+            io_manager_key="parquet_io_manager"
+        ),
+        "core_eia930__assn_balancing_authority_region": AssetOut(
+            io_manager_key="pudl_io_manager"
```

Comment:
Writing the association tables to the DB is useful because it ends up with a BA code FK relationship, and that identified some new BAs that needed to be added to the coding table (Mexico & Canada BAs)